### PR TITLE
Add order relation to gift card event model

### PR DESCRIPTION
**Pull request by IKarbowiak**
Extend `GiftCardEvent` with `order` field.
Rewrite order data from `parameters` to relation.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6142 Add order relation to GiftCardEvent model](https://app.clickup.com/t/2549495/SALEOR-6142)

**Comment by patrys** [comment]
Task linked: [SALEOR-5849 Migrate order IDs to UUID](https://app.clickup.com/t/2549495/SALEOR-5849)

File: saleor/graphql/giftcard/filters.py [hunk 1]
```diff
@@ -128,14 +130,31 @@ def filter_events_by_type(events: List[models.GiftCardEvent], type_value: str):
...
 def filter_events_by_orders(events: List[models.GiftCardEvent], order_ids: List[str]):
-    _, order_pks = resolve_global_ids_to_primary_keys(order_ids, "Order")
+    order_pks = _get_order_pks(order_ids)
+
     filtered_events = []
     for event in events:
-        if str(event.parameters.get("order_id")) in order_pks:
+        if event.order_id in order_pks:
             filtered_events.append(event)
     return filtered_events
```
Comment (fowczarek) [inline saleor/graphql/giftcard/filters.py:139]: Maybe we could rewrite it to don't use `for` loop.  We can use `Exists` on `OrderQS` with `Q(id__in=pks) | (Q(use_old_id=True) & Q(number__in=old_pks))`  lookup.

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/giftcard/filters.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/graphql/giftcard/filters.py:139]: In this case, we will add an additional query, as the `events` are in the list.

Comment (fowczarek) [reply saleor/graphql/giftcard/filters.py:139]: Yup. It's my mistake. My assumption is that `events` is QS. Let's leave it as it is now.

**Review by fowczarek — APPROVED** [review]

**Review by jakubkuc — APPROVED** [review]

**Pull request merged**